rust/theoros/src/storage/checkpoints.rs

```diff
@@ -0,0 +1,83 @@
+use std::collections::{BTreeMap, HashMap};
+use std::sync::Arc;
+
+use starknet::core::types::Felt;
+use tokio::sync::RwLock;
+
+use crate::types::hyperlane::{DispatchEvent, SignedCheckpointWithMessageId};
+
+/// Mapping between messages nonces and their corresponding Event.
+#[derive(Clone, Default)]
+pub struct UnsignedCheckpointsStorage(Arc<RwLock<BTreeMap<u32, DispatchEvent>>>);
```

Comment:
same here, also because we only care about getting stuff in descending nonces order probably [ordermap](https://github.com/indexmap-rs/ordermap) is better suited

Reply:
see my other comment, not sure then